### ISSUE TITLE
KroneckerProduct doesn’t inherit eltype

### ISSUE THREAD

**Issue by mcabbott** [MichielStock/Kronecker.jl#53]
This package seems to be subtyping `AbstractArray` in a slightly nonstandard way, which causes various operations to land up with an abstract eltype like `Number`:
```
julia> k = kronecker(rand(1,2), 2)
1×4 KroneckerPower{Array{Float64,2},2}:
 0.0352727  0.15181  0.15181  0.653376

julia> k isa AbstractArray
true

julia> eltype(k)
Float64

julia> ndims(k)
2

julia> k isa AbstractArray{Float64, 2}
false

julia> supertype(typeof(k)) # without type parameters
AbstractKroneckerProduct

julia> supertype(supertype(typeof(rand(2,2))))
AbstractArray{Float64,2}

julia> k[:,1]
1-element Array{Number,1}:
 0.03527270456415509

julia> permutedims(k)
4×1 Array{Number,2}:
 0.8307185916636805 
 0.23112342316803244
 0.8816855796908125 
 0.24530351358565958
```

**Comment by MichielStock** [comment]
Thanks for bringing this to my attention. Initially, I did not use inheritance of abstract types. In the branch [types](https://github.com/MichielStock/Kronecker.jl/tree/types) I have solved this problem and added the appropriate unit tests. When checks are completed I will merge, bump the version number and close this.

**Comment by MichielStock** [comment]
Closed by 0e62b2a
